### PR TITLE
test_filters: rewrite test cases in criterion

### LINKED ISSUE #2572: convert ut test_filters_in_list to criterion
Use Criterion to rewrite this unit test.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
fixes #2572

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

File: lib/filter/tests/test_filters_in_list.c [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #define LIST_FILE_DIR "%s/lib/filter/tests/filters-in-list/"
 
 static MsgFormatOptions parse_options;
+char *top_srcdir = NULL;
```
Comment (furiel) [inline lib/filter/tests/test_filters_in_list.c]: Wouldn't be `char *top_srcdir = TOP_SRCDIR;` sufficient here? Do we need strdup/free in setup/teardown?

Comment (mehul-m-prajapati) [reply lib/filter/tests/test_filters_in_list.c]: Hi @furiel ,

Yes, the static allocation is enough. I will change the code.

**Comment by furiel** [comment]
@kira-syslogng ok to test

File: lib/filter/tests/test_filters_in_list.c (hunk 1 above)
Comment (furiel) [reply lib/filter/tests/test_filters_in_list.c]: Thanks!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**